pipeline/gitlab-ci/dmc.yml

```diff
@@ -0,0 +1,29 @@
+---
+build_dmc:
+  stage: docker_management_container
+  image: docker:git
+  services:
+    - name: docker:dind
+      alias: docker
+  variables:
+    # renovate: datasource=github-releases depName=docker-management-container
+    GITHUB_DMC_TAG: 3.1.0
+    DMC_IMAGE: $CI_REGISTRY_IMAGE/${DMC_NAME}
+  script:
+    # get date for tagging the image
+    - export DATE=$(date +"%F")
+    # get DMC from GitHub
+    - git clone --branch ${GITHUB_DMC_TAG} --depth 1 https://github.com/T-Systems-MMS/docker-management-container $CI_PROJECT_DIR/docker/docker-management-container
+    # create Dockerfile with template from submodule
+    - sh $CI_PROJECT_DIR/docker/docker-management-container/render.sh $CI_PROJECT_DIR/docker/${DMC_NAME}
+    # build and tag image
+    - docker build --label dmc-version=${GITHUB_DMC_TAG} -t ${DMC_IMAGE} $CI_PROJECT_DIR/docker/${DMC_NAME}
+    - docker tag $DMC_IMAGE:latest $DMC_IMAGE:${DATE}
+    # push images
+    - docker push --all-tags ${DMC_IMAGE}
+  rules:
+    - if: $CI_COMMIT_BRANCH == "master" || $CI_COMMIT_BRANCH == "main"
+    - changes:
+      - docker/${DMC_NAME}/*
+  extends:
+    - .docker_login
```

Comment:
This doesn't exist everywhere. I'd say we just run `docker login -u $CI_REGISTRY_USER -p $CI_REGISTRY_PASSWORD $CI_REGISTRY` as a step in the script.

Reply:
I would the extends stay as it is, because some user wil put der dmc image in an other Docker Registry than gitlab. I will add at this to the README.MD for full usage documentation.